### ISSUE TITLE
test_warn macro failure?

### ISSUE THREAD

**Issue by richardreeve** [JuliaCI/PackageEvaluator.jl#159]
I'm puzzled by a PackageEvaluator [failure](http://pkg.julialang.org/?pkg=Diversity&ver=0.6) under Julia 0.6. The [log](http://pkg.julialang.org/logs/Diversity_0.6.log) suggests a bunch of [deprecation warnings](https://github.com/richardreeve/Diversity.jl/blob/master/test/test_Diversity.jl) that I check for are failing:

```
INFO: Running tests...
    * Testing Diversity.jl ...
Deprecations: Test Failed
  Expression: (Base.Test.ismatch_warn)("deprecated", (Base.Test.readstring)(#16#fname))
Stacktrace:
 [1] macro expansion at ./test.jl:429 [inlined]
 [2] macro expansion at /home/vagrant/.julia/v0.6/Diversity/test/test_Diversity.jl:0 [inlined]
 [3] macro expansion at ./test.jl:853 [inlined]
 [4] anonymous at ./<missing>:?
...
Deprecations: Test Failed
  Expression: (Base.Test.ismatch_warn)("deprecated", (Base.Test.readstring)(#258#fname))
Stacktrace:
 [1] macro expansion at ./test.jl:429 [inlined]
 [2] macro expansion at /home/vagrant/.julia/v0.6/Diversity/test/test_Diversity.jl:0 [inlined]
 [3] macro expansion at ./test.jl:853 [inlined]
 [4] anonymous at ./<missing>:?
Test Summary: | Pass  Fail  Total
Deprecations  |    1    23     24
ERROR: LoadError: LoadError: Some tests did not pass: 1 passed, 23 failed, 0 errored, 0 broken.
while loading /home/vagrant/.julia/v0.6/Diversity/test/test_Diversity.jl, in expression starting on line 11
while loading /home/vagrant/.julia/v0.6/Diversity/test/runtests.jl, in expression starting on line 16
```

However, they work fine both on my computer:

```
INFO: Running tests...
    * Testing Diversity.jl ...
Test Summary: | Pass  Total
Deprecations  |   24     24
```

and using [Travis](https://travis-ci.org/richardreeve/Diversity.jl/builds/220371712) and [Appveyor](https://ci.appveyor.com/project/richardreeve/diversity-jl/build/1.0.74) via github for the same version of Diversity (v0.3.0) that's failing for PackageEvaluator. I can reproduce the error using vagrant on my machine, but I can't see why it would be failing... does anyone have any suggestions? The test_warn macro doesn't exist on v0.5, so it's irrelevant that it works there.

**Comment by tkelman** [comment]
According to http://pkg.julialang.org/detail/Diversity.html it started failing sometime between the 6th and the 13th (for various annoying reasons PkgEval didn't successfully finish a run in that intervening week). So this may be a change in Base, and could be worth bisecting to find what it was, if it's an inadvertent breaking change that we should do something about (we're supposed to be in feature freeze for 0.6).

What `versioninfo()` of Julia were you testing on locally?

**Comment by richardreeve** [comment]
I'm using 0.6.0-pre.beta.122 at the moment:

```
Julia Version 0.6.0-pre.beta.122
Commit aaec5e228e (2017-04-13 02:34 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin13.4.0)
  CPU: Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, skylake)
```

The date range you cite is when v0.3.0 was added to [METADATA](https://github.com/JuliaLang/METADATA.jl/pull/8751). Before that there was only 0.2.8 to look at...

**Comment by tkelman** [comment]
Oh right. Then I'd bisect on your package?

**Comment by richardreeve** [comment]
But the problem is I can't get anything to fail except when I'm running it through vagrant, and I can only run vagrant on released packages... I edited `Vagrantfile` and `setup.sh` to only run my test successfully, but I failed to get it to run on a clone of a branch where I'm making specific edits to try to understand what's happening:

```
timeout -s 9 2000s julia -e "Pkg.clone(\"https://github.com/richardreeve/Diversity.jl.git\", \"R\")" 2>&1 | tee PKGEVAL_${PKGNAME}_add.log
```

When I run either v0.3.0 or master on my machine or through github on travis / appveyor everything succeeds.

Despite that, it seems pretty clear that the test_warn macro isn't returning a warning as it does on my computer - I wondered whether vagrant was doing something "clever" with stderr which was making these tests fail?

For instance, without really knowing what's going on [src/preptest.jl](https://github.com/JuliaCI/PackageEvaluator.jl/blob/master/src/preptest.jl) lines 42-43 seems to be pushing stderr into stdio (2>&1), though I can't see how that would effect something inside Julia tbh.

**Comment by tkelman** [comment]
Latest master of Julia does fail with Diversity 0.3.0 for me, so I think it was a Julia change rather than a package change.

**Comment by richardreeve** [comment]
Ah, okay. I just downloaded the latest version from https://julialang.org/downloads/ - how do I get the 'real' latest version.

**Comment by tkelman** [comment]
Just bisected to https://github.com/JuliaLang/julia/pull/21367, so it was a regression just a few days ago.

**Comment by richardreeve** [comment]
I've successfully downloaded:

```
Julia Version 0.6.0-pre.beta.145
Commit f824e4bfd9 (2017-04-15 10:34 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin13.4.0)
  CPU: Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, skylake)
```

and the test still works on my (macOS) machine! That's the exact version that is on my copy of vagrant which fails (albeit under linux). If you think that really is the problem though, id there anything that can be done to fix it?

**Comment by tkelman** [comment]
It could easily be a Linux-vs-Mac difference, these backtrace issues can be platform dependent. Maybe you hit the same deprecation message earlier than this particular test?

**Comment by richardreeve** [comment]
No, I'm afraid not. I only discovered the test_warn macro existed a few days ago. In any event, I've pulled those tests from the linux testing and released v0.3.1 of my package - JuliaLang/METADATA.jl#8866 - with some other minor fixes so it should pass v0.6 tests for the time being.

**Comment by IainNZ** [comment]
Closing due to age.
